### ISSUE TITLE
Linux Steam no longer launches from Lutris since 0.4.15/0.4.16 update

### ISSUE THREAD

**Issue by DaDummy** [lutris/lutris#869]
Lutris 0.4.16 on Kubuntu Bionic Beaver (18.04) here.

I've got Lutris with wine, wine-steam, steam and some Linux games on my system.

Since 0.4.15 or 0.4.16 I'm having trouble running Steam and Steam games through Lutris - the process dies instantly, which it didn't before. I haven't changed anything except for updating Lutris through apt. Running Steam from Dolphin/bash still works as expected.

As far as I have tested some Linux games still work others do not.
As far as I have tested wine games still work as before.
The wine-steam runner and all installed wine-steam games have disappeared from Lutris with the update, so I cannot test if those still work. (My wine-steam setup was a little unconventional since I did want my wine-steam in a win64 prefix which I didn't know how to configure through Lutris back then.)

Please let me know if there is any way I can narrow down the cause of that problem.

**Comment by DaDummy** [comment]
Some additional info that might be relevant:
I have lutris installed from the Lutris apt repo for Ubuntu 18.04.

wine-development and steam:i386 are installed from Ubuntu repos, wine-stable is not installed.

**Comment by DaDummy** [comment]
I ran a git bisect and this appears to be the commit that broke Linux-Steam for my system:

```
d8a664f444c2d9d9be1c6d74890bf18a93bfbd62 is the first bad commit
commit d8a664f444c2d9d9be1c6d74890bf18a93bfbd62
Author: Travis Nickles <nickles.travis@gmail.com>
Date:   Wed Apr 18 21:39:23 2018 -0500

    Fixed winesteam data download, winesteam auto prefix workaround, winesteam install
    
    Use proper runner config data when downloading game data from Steam
    otherwise no prefix is detected (was attempting to use linux data).
    For default winesteam install, use default_arch if auto is set for
    the prefix architecture. Steam for Windows install can get stuck when
    output from stdout is not read. If watch is not set, just close the
    stdout stream.

:040000 040000 0d50bf3d56711cb6cc77ba7892bd9e0a8592dc76 63ba9b799b84109891fc6894b4ec8aed92941fdd M      lutris
```

**Comment by DaDummy** [comment]
As stated in the title: Linux Steam.

I just mentioned wine as the presence of wine packages changes the behavior of Lutris now according to the changelog. I have further pinpointed the cause of the issue and am preparing a patch. That fixes the issue in master for me. It's a (very) partial revert of the commit bisect identified as the cause.

**Comment by strycore** [comment]
The reason asked is that the commit returned by the bisect seems specific to Wine Steam. I'm setting up a test environment where I can reproduce that more easily.

**Comment by DaDummy** [comment]
I've submitted a pull request that fixes the issue for my system: #870.

I was just as surprised as you were that this commit seemed to be responsible. Sadly the commit message suggest that without this code wine-steam did not work/install for him.

My fix probably needs to be followed up or refactored to address his issue in a more specific way that does not (negatively) affect native Linux applications.

**Comment by telanus** [comment]
I don't get the problems on my system. Kubuntu 18.04 (x86_64 Linux 4.15.0-20-generic). Tried it with an installed game and it launched it without problem, and no warning written to the terminal

**Comment by DaDummy** [comment]
The issue appears to be limited to applications launched indirectly through a shell script on my system.

I've reset per application and global settings that could affect this now, but the problem persists. I'll keep looking for the cause...

**Comment by DaDummy** [comment]
The issue only occurs if process monitor is disabled for the application through either application or lutris preferences. All other settings do not appear to have an effect. The issue is further caused by game_process.stdout.close() sending a SIGPIPE to bash which on my system reacts to that by exiting.

**Comment by strycore** [comment]
The process monitor should no longer be set to disabled, setting this to a milestone far away, this has suddently dropped a lot in priority :D

**Comment by DaDummy** [comment]
Why is that? I don't want the stop-button on Lutris to kill my voice chat when I use it to kill a defunct League of Legends client. Is there a new feature that replaces this functionality? Also if it is not supported to disable the process monitor, why is it still included in the UI? Are there other unsupported options in the menu?

**Comment by strycore** [comment]
The option was added because there was an issue with the process monitor back then, I believe this is now fixed (I've been able to reproduce an fix the issue with Epic Launcher).

The voice chat aspect is quite different, does the LoL client require force quitting the game to exit properly? And is your voice chat application Wine based as well?

**Comment by DaDummy** [comment]
Well nevermind. Disabling the process monitor does not help with the stop-button anymore. I guess I'll just properly implement this feature and supply a new PR.

A new release with the process monitor-option removed might still be a good idea to avoid other ppl running into the trouble I had.

**Comment by strycore** [comment]
Before removing the option, I'd like to make sure the process monitor now works consistently. I can also take another look at your PR now that I know what's causing the issue.

**Comment by DaDummy** [comment]
I believe the problem this issue was about has been fixed with the merge of #870
